sensor/kubernetes/certrefresh/securedcluster_tls_issuer.go

```diff
@@ -0,0 +1,138 @@
+package certrefresh
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/certificates"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/securedcluster"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+var _ common.SensorComponent = (*securedClusterTLSIssuerImpl)(nil)
+
+// NewSecuredClusterTLSIssuer creates a sensor component that will keep the Secured Cluster certificates
+// up to date, using the retry parameters in tls_issuer_common.go
+func NewSecuredClusterTLSIssuer(
+	k8sClient kubernetes.Interface,
+	sensorNamespace string,
+	sensorPodName string,
+) common.SensorComponent {
+	msgToCentralC := make(chan *message.ExpiringMessage)
+	msgFromCentralC := make(chan *central.IssueSecuredClusterCertsResponse)
+	return &securedClusterTLSIssuerImpl{
+		sensorNamespace:              sensorNamespace,
+		sensorPodName:                sensorPodName,
+		k8sClient:                    k8sClient,
+		msgToCentralC:                msgToCentralC,
+		msgFromCentralC:              msgFromCentralC,
+		certRefreshBackoff:           certRefreshBackoff,
+		getCertificateRefresherFn:    newCertificatesRefresher,
+		getServiceCertificatesRepoFn: securedcluster.NewServiceCertificatesRepo,
+		certRequester:                securedcluster.NewCertificateRequester(msgToCentralC, msgFromCentralC),
+	}
+}
```

Comment:
If we look at this block, then it should be trivial to refactor it in a way, that the `certRequester` would create the channel and return it. It would also know best when to close it (as currently this channel is not being closed).